frontend/src/metabase/query_builder/components/filters/modals/InlineCategoryPicker/InlineCategoryPicker.unit.spec.tsx

```diff
@@ -0,0 +1,337 @@
+};
+
+const largeCategoryField = new Field({
+  database_type: "test",
+  semantic_type: "type/Category",
+  table_id: 8,
+  name: "large_category_field",
+  has_field_values: "list",
+  values: new Array(MAX_INLINE_CATEGORIES + 1).fill(null).map(turtleFactory),
+  dimensions: {},
+  dimension_options: [],
+  effective_type: "type/Text",
+  id: 138,
+  effective_type: "type/Text",
+  metadata,
+});
+
+const emptyCategoryField = new Field({
+  database_type: "test",
+  semantic_type: "type/Category",
+  table_id: 8,
+  name: "empty_category_field",
+  has_field_values: "list",
+  values: [],
+  dimensions: {},
+  dimension_options: [],
+  effective_type: "type/Text",
+  id: 139,
+  effective_type: "type/Text",
+  metadata,
+});
+
+metadata.fields[smallCategoryField.id] = smallCategoryField;
+metadata.fields[largeCategoryField.id] = largeCategoryField;
+metadata.fields[emptyCategoryField.id] = emptyCategoryField;
+
+const card = {
+  dataset_query: {
+    database: 5,
+    query: {
+      "source-table": 8,
+    },
+    type: "query",
+  },
+  display: "table",
+  visualization_settings: {},
+};
+
+const question = new Question(card, metadata);
+const query = question.query();
+const smallDimension = smallCategoryField.dimension();
+const largeDimension = largeCategoryField.dimension();
+const emptyDimension = emptyCategoryField.dimension();
+
+describe("InlineCategoryPicker", () => {
+  it("should render an inline category picker", () => {
+    const testFilter = new Filter(
+      ["=", ["field", smallCategoryField.id, null], undefined],
+      null,
+      query,
+    );
+    const changeSpy = jest.fn();
+    const fetchSpy = jest.fn();
+
+    render(
+      <InlineCategoryPickerComponent
+        query={query}
+        filter={testFilter}
+        newfilter={testFilter}
+        handleChange={changeSpy}
+        fieldValues={smallCategoryField.values}
+        fetchFieldValues={fetchSpy}
+        dimension={smallDimension}
+        handleClear={changeSpy}
+      />,
+    );
+
+    screen.getByTestId("category-picker");
+    smallCategoryField.values.forEach(value => {
+      screen.getByText(value);
+    });
+  });
+
+  it("should render a loading spinner while loading", async () => {
+    const testFilter = new Filter(
+      ["=", ["field", emptyCategoryField.id, null], undefined],
+      null,
+      query,
+    );
+    const changeSpy = jest.fn();
+    const fetchSpy = jest.fn();
+
+    render(
+      <InlineCategoryPickerComponent
+        query={query}
+        filter={testFilter}
+        newfilter={testFilter}
+        handleChange={changeSpy}
+        fieldValues={emptyCategoryField.values}
+        fetchFieldValues={fetchSpy}
+        dimension={emptyDimension}
+        handleClear={changeSpy}
+      />,
+    );
+    screen.getByTestId("loading-spinner");
+    await waitFor(() => expect(fetchSpy).toHaveBeenCalled());
+  });
+
+  it("should render a warning message on api failure", async () => {
```

Comment:
It's the lack of field values that causes the fetch. The difference is when we test it. before the fetch, we get a loading spinner, after the fetch we get an error (because we haven't mocked the api)